packages/utils/src/miscellaneous.ts

```diff
@@ -0,0 +1,24 @@
+const TAG = 'utils/src/miscellaneous'
+
+type InFunction = (...params: any) => Promise<any>
+
+export const retryAsync = async (
+  inFunction: InFunction,
+  tries: number,
+  params: any,
+  delay = 100
+) => {
+  let saveError
+  for (let i = 0; i < tries + 1; i++) {
+    try {
+      // it awaits otherwaise it'd always do all the retries
```

Comment:
typo: otherwise
Can you also mention that the inFunction *must* throw an error on failures otherwise this mechanism doesn't work